docs/libcurl/opts/CURLOPT_V4_SIGNATURE.3

```diff
@@ -0,0 +1,109 @@
+.\" *
+.\" * This software is licensed as described in the file COPYING, which
+.\" * you should have received as part of this distribution. The terms
+.\" * are also available at https://curl.haxx.se/docs/copyright.html.
+.\" *
+.\" * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+.\" * copies of the Software, and permit persons to whom the Software is
+.\" * furnished to do so, under the terms of the COPYING file.
+.\" *
+.\" * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+.\" * KIND, either express or implied.
+.\" *
+.\" **************************************************************************
+.\"
+.TH CURLOPT_V4_SIGNATURE 3 "03 Jun 2020" "libcurl 7.72.0" "curl_easy_setopt options"
+.SH NAME
+CURLOPT_V4_SIGNATURE \- V4 signature (used by some cloud providers)
+.SH SYNOPSIS
+.nf
+#include <curl/curl.h>
+
+CURLcode curl_easy_setopt(CURL *handle, CURLOPT_V4_SIGNATURE,
+                          char *providers_infos);
+.SH DESCRIPTION
+provides AWS V4 signature authentication on HTTPS header
+
+The provider argument is a string that is merged to some authentication
+parameters use by the algorithm.
+It's used by "Algorithm", "date", "request type", "signed headers" arguments,
+
+Example with "Test:Try":
+When curl will do the algorithm, it will Generate:
+"TEST-HMAC-SHA256" for "Algorithm"
+"x-try-date" and "X-Try-Date" for "date"
+"test4_request" for "request type"
+"SignedHeaders=content-type;host;x-try-date" for "signed headers"
+
+If you use just "test", instead of "test:try",
+test will be use for every strings generated
+
+Small explication of what the algorithm does:
+
+require arguments:
+- The previously mentioned arguments that are created from prefix
+- some arguments extracted from the URL:
+        - "api type"
+        - "cloud region"
+        - hostname
+        - URL path
+- The current date (get with Curl_gmtime)
+- The access key and the secret key (store in CURLOPT_USERPWD)
+- HTTP content type header
+
+
+Performed Operations:
+- make "canonical header" using "content type", the hostname, and the date
+- hash the post data
+- make "canonical request" using "request type", URL path,
+  get data (if any), "canonical header", "signed header"
+  and post data sha256 hash
+- hash "canonical request"
+- make "credential scope" using date, region, "api type" and "request type"
+- make "string to sign" using "Algorithm",
+  the date, "credential scope" and "canonical request" hash
+- compute hmac from date, using "secret key" as key.
+- compute hmac from region, using above hmac as key
+- compute hmac from "api type", using above hmac as key
+- compute hmac from "request type", using above hmac as key
+- compute hmac from "string to sign" using above hmac as key
+- create Authorization HTTP header using "Algorithm", access key,
+  "credential scope", "signed header" and above hash
+- Push the "date" and the Authorization to the HTTP header
```

Comment:
Is this very elaborate description really necessary?

Reply:
maybe not, 
I'm just not good at finding the level of description.

Reply:
You don't need to describe how the auth works here, that's out of scope for the man page.